src/utils/fxa.ts

```diff
@@ -47,34 +46,47 @@ async function destroyOAuthToken(
 
   try {
     const response = await fetch(tokenUrl, tokenOptions);
-    if (!response.ok)
-      throw new InternalServerError(`bad response: ${response.status}`);
-    await response.json();
-    return;
+    const responseJson = await response.json();
+    if (!response.ok) throw responseJson;
```

Comment:
Hm, I think this may be a bit hard to debug from logs... wouldn't this throw `responseJson` (which is a `Response` object but not an `Error`) and then fail the `e instanceof Error` test below and this would end up returning `false` without any insight into why it failed?

Reply:
You're right that it'll fail the `instanceof Error` check, but it's not a `Response` object - that's `response`. `responseJson` will be the parsed response body.

However, also note that if the error response body is _not_ valid JSON, that this will also throw an error, in which case we might want to move it until _after_ the `response.ok` check.

(It's also not clear to me why the original code called `await response.json()` in the first place, only to then discard the return value.)

Reply:
I first saw this pattern in the next-auth documentation, but now thinking more about it, perhaps we need some modifications. When failed, there are essentially 3 possible scenarios for line 49 `const responseJson = await response.json();`:

1. Failed call returns a valid JSON (fxa does this pretty well, so I'd say we should be able to expect this when dealing with fxa)
2. Empty response, error thrown for `can't parse` (hopefully not with fxa)
3. Some other response (html maybe), error thrown for `can't parse` (again, not likely with fxa)

It's a good point that `throw json` may not be an `Error` object. I think we should handle everything in the `catch()` below